src/devices/Nrf24l01/Nrf24l01.cs

```diff
@@ -0,0 +1,773 @@
+        /// Initialize
+        /// </summary>
+        private void Initialize(PinNumberingScheme pinNumberingScheme, OutputPower outputPower, DataRate dataRate, byte channel)
+        {
+            _gpio = new GpioController(pinNumberingScheme);
+            _gpio.OpenPin(_ce, PinMode.Output);
+            _gpio.OpenPin(_irq, PinMode.Input);
+            _gpio.RegisterCallbackForPinValueChangedEvent(_irq, PinEventTypes.Falling, Irq_ValueChanged);
+
+            Thread.Sleep(50);
+
+            // Details in the datasheet P53
+            _gpio.Write(_ce, PinValue.Low);
+            Write(Command.NRF_FLUSH_TX, Register.NRF_NOOP, _empty);
+            Write(Command.NRF_FLUSH_RX, Register.NRF_NOOP, _empty);
+            Write(Command.NRF_W_REGISTER, Register.NRF_CONFIG, 0x3B);
+            _gpio.Write(_ce, PinValue.High);
+
+            SetAutoAck(false);
+            SetChannel(channel);
+            SetOutputPower(outputPower);
+            SetDataRate(dataRate);
+            SetRxPayload(_packetSize);
+        }
+
+        /// <summary>
+        /// Initialize nRF24L01 Pipe
+        /// </summary>
+        private void InitializePipe()
+        {
+            Pipe0 = new Nrf24l01Pipe(this, 0);
+            Pipe1 = new Nrf24l01Pipe(this, 1);
+            Pipe2 = new Nrf24l01Pipe(this, 2);
+            Pipe3 = new Nrf24l01Pipe(this, 3);
+            Pipe4 = new Nrf24l01Pipe(this, 4);
+            Pipe5 = new Nrf24l01Pipe(this, 5);
+        }
+
+        /// <summary>
+        /// Set nRF24L01 Receive Packet Size (All Pipe)
+        /// </summary>
+        /// <param name="payload">Size, from 0 to 32</param>
+        internal void SetRxPayload(byte payload)
+        {
+            if (payload > 32 || payload < 0)
+            {
+                throw new ArgumentOutOfRangeException("payload", "payload needs to be in the range of 0 to 32!");
+            }
+
+            _gpio.Write(_ce, PinValue.Low);
+
+            Span<byte> writeData = stackalloc byte[1] { payload };
+
+            Write(Command.NRF_W_REGISTER, Register.NRF_RX_PW_P0, writeData);
+            Write(Command.NRF_W_REGISTER, Register.NRF_RX_PW_P1, writeData);
+            Write(Command.NRF_W_REGISTER, Register.NRF_RX_PW_P2, writeData);
+            Write(Command.NRF_W_REGISTER, Register.NRF_RX_PW_P3, writeData);
+            Write(Command.NRF_W_REGISTER, Register.NRF_RX_PW_P4, writeData);
+            Write(Command.NRF_W_REGISTER, Register.NRF_RX_PW_P5, writeData);
+
+            _gpio.Write(_ce, PinValue.High);
+        }
+
+        /// <summary>
+        /// Set nRF24L01 Receive Packet Size
+        /// </summary>
+        /// <param name="pipe">Pipe, form 0 to 5</param>
+        /// <param name="payload">Size, from 0 to 32</param>
+        internal void SetRxPayload(byte pipe, byte payload)
+        {
+            if (payload > 32 || payload < 0)
+            {
+                throw new ArgumentOutOfRangeException("payload", "payload needs to be in the range of 0 to 32!");
```

Comment:
nit: use `nameof(payload)` when referring to argument names. Also don't use exclamation points. So, in this case:

``` C#
throw new ArgumentOutOfRangeException(nameof(payload), $"{nameof(payload)} needs to be in the range of 0 to 32.");
```

Reply:
the reasoning for that is so that tools can work correctly when refactoring (i.e. VS will rename variable correct) - updating string won't happen automatically while nameof works fine